pkg/template/api/validation/validation.go

```diff
@@ -12,7 +12,7 @@ import (
 	unversionedvalidation "k8s.io/kubernetes/pkg/api/unversioned/validation"
 )
 
-var parameterNameExp = regexp.MustCompile(`^[a-zA-Z0-9\_]+$`)
+var parameterNameExp = regexp.MustCompile(`^[a-zA-Z0-9_]+$`)
```

Comment:
Both work identically, but I believe the \ is a syntax error which is being tolerated by the Golang regex parser.